ci-operator/step-registry/vsphere/provision/bastionhost/vsphere-provision-bastionhost-commands.sh

```diff
@@ -0,0 +1,141 @@
+govc vm.change -vm ${vm_folder}/${bastion_name} -e "guestinfo.ignition.config.data=${bastion_ignition_base64}"
+govc vm.power -on ${vm_folder}/${bastion_name}
+
+loop=10
+while [ ${loop} -gt 0 ]; do
+  bastion_ip=$(govc vm.info -json ${vm_folder}/${bastion_name} | jq -r .VirtualMachines[].Summary.Guest.IpAddress)
+  if [ "x${bastion_ip}" == "x" ]; then
+    loop=$(( loop - 1 ))
+    sleep 30
+  else
+    break
+  fi
+done
+
+if [ "x${bastion_ip}" == "x" ]; then
+  echo "Unabel to get ip of bastion host instance ${bastion_name}!"
+  exit 1
+fi
+
+echo "${vm_folder}/${bastion_name}" > "${SHARED_DIR}/bastion_host_path"
+echo "Bastion host created."
+
+#Create dns for bastion host
+if [[ "${REGISTER_MIRROR_REGISTRY_DNS}" == "yes" ]]; then
+  echo "$(date -u --rfc-3339=seconds) - Creating bastion host DNS..."
+  if [[ -f "${SHARED_DIR}"/basedomain.txt ]]; then
+    base_domain=$(<"${SHARED_DIR}"/basedomain.txt)
+  else
+    echo "Unable to find basedomain.txt in SHARED_DIR, set default value..."
+    base_domain="vmc-ci.devcluster.openshift.com"
+  fi
+  export AWS_SHARED_CREDENTIALS_FILE=/var/run/vault/vsphere/.awscred
+  export AWS_MAX_ATTEMPTS=50
+  export AWS_RETRY_MODE=adaptive
+
+  bastion_host_dns="${bastion_name}.${base_domain}"
+  bastion_hosted_zone_id="$(aws route53 list-hosted-zones-by-name \
+            --dns-name "${base_domain}" \
+            --query "HostedZones[? Config.PrivateZone != \`true\` && Name == \`${base_domain}.\`].Id" \
+            --output text)"
+  echo "${bastion_hosted_zone_id}" > "${SHARED_DIR}/bastion-hosted-zone.txt"
+ 
+  dns_create_str=""
+  dns_delete_str=""
+  dns_target='"TTL": 60,"ResourceRecords": [{"Value": "'${bastion_ip}'"}]'
+  upsert_str="{\"Action\": \"UPSERT\",\"ResourceRecordSet\": {\"Name\": \"${bastion_host_dns}.\",\"Type\": \"A\",$dns_target}}"
+  delete_str="{\"Action\": \"DELETE\",\"ResourceRecordSet\": {\"Name\": \"${bastion_host_dns}.\",\"Type\": \"A\",$dns_target}}"
+  dns_create_str="${upsert_str},${dns_create_str}"
+  dns_delete_str="${delete_str},${dns_delete_str}"
+
+  cat > "${SHARED_DIR}"/bastion-host-dns-create.json <<EOF
+{"Comment": "Create public OpenShift DNS records for bastion host on vSphere","Changes": [${dns_create_str::-1}]}
+EOF
+
+  cat > "${SHARED_DIR}"/bastion-host-dns-delete.json <<EOF
+{"Comment": "Delete public OpenShift DNS records for bastion host on vSphere","Changes": [${dns_delete_str::-1}]}
+EOF
+
+  id=$(aws route53 change-resource-record-sets --hosted-zone-id "$bastion_hosted_zone_id" --change-batch file:///"${SHARED_DIR}"/bastion-host-dns-create.json --query '"ChangeInfo"."Id"' --output text)
+  echo "Waiting for DNS records to sync..."
+  aws route53 wait resource-record-sets-changed --id "$id"
+  echo "DNS records created."
+
+  MIRROR_REGISTRY_URL="${bastion_host_dns}:5000"
+  echo "${MIRROR_REGISTRY_URL}" > "${SHARED_DIR}/mirror_registry_url"
+fi
+
+echo "Removing temporary ova template if existing..."
```

Comment:
To reuse bastion ova template, do you think it is deserved keep the vm template stay there?

Reply:
I though it may impact ova image processing in ref `upi-conf-vsphere-ova`, but I renamed it with post fix "-bastion", should not impact, agree to keep it, will update.